### DISCUSSION TITLE
Condition on target rate sensor

### DISCUSSION THREAD

**Discussion by ccristal** [BottlecapDave/HomeAssistant-OctopusEnergy#424] [Ideas]
Hi! I have the following scenario: I want the sensor to turn on on the cheapest 2 hours for which the current_average_cost is lower than my export rate.
This is not the same thing as a sensor that turns on on the cheapest two hours, and then setting the condition on top of it in an automation. For example, suppose that my appliance runs for 2 hours, and I have rates as in the following table:

| Time | Import rate | Export Rate | Sensor | Condition | Sensor AND Condition | Sensor with Condition |
| --- | --- | --- | --- | --- | --- | --- |
| 00:00 |-0.0215|0.15|on|on|on|on |
| 00:30 |-0.0220|0.15|on|on|on|on  |
| 01:00 |-0.0225|0.01|on|off|off|off |
| 01:30 |-0.0230|0.15|on|on|on|on |
| 02:00 |-0.0235|0.15|off|on|off|on |
| 02:30 |-0.0240|0.15|off|on|off|on|

You can see that evaluating a condition on top of the sensor only activates 3 slots, whereas having a sensor with a built-in condition activates 4, which is what I need to run my appliance at the cheapest possible cost.

Hope it makes sense... :-)

**Comment by BottlecapDave** [comment]
Hello. I'm not really following your table. Are you looking for a continuous 2 hours, or intermittent 2 hours? Which slots would you expect the sensor to come on for?

**Comment by ccristal** [comment]
Closing as it doesn't make sense. Replaced by #430 .
